public/users/totp.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * Postfix Admin
+ *
+ * LICENSE
+ * This source file is subject to the GPL license that is bundled with
+ * this package in the file LICENSE.TXT.
+ *
+ * Further details on the project are available at http://postfixadmin.sf.net
+ *
+ * @version $Id$
+ * @license GNU GPL v2 or later.
+ *
+ * File: totp.php
+ * Used by users to change their mailbox (and login) totp.
+ * Template File: totp.tpl
+ *
+ *
+ * Form POST \ GET Variables:
+ *
+ * fPassword_current
+ * fTOTP_code
+ * fTOTP_secret
+ */
+
+require_once('../common.php');
+
+$smarty = PFASmarty::getInstance();
+$smarty->configureTheme('../');
+
+$username                                    = authentication_get_username();
+$pPassword_password_current_text             = "";
+$pTOTP_code_text                             = "";
+$pTOTP_secret_text                           = "";
+$pTOTP_now                                   = "";
+$pPassword_password_text                     = "";
+$pQR_raw               	                     = "";
+
+if (authentication_has_role('admin')){
+    $totppf = new TotpPf('admin');
+    $login  = new Login('admin');
+    $admin = true;
+}else{
+    $totppf = new TotpPf('mailbox');
+    $login  = new Login('mailbox');
+    $admin = false;
+}
+
+
+// Create new OTP-object 
```

Comment:
When TOTP is already configured, this form could be hidden until the users explicitly uses a button "Change TOTP" (or similar).

Reply:
True, done.